### PR TITLE
Support walrus file apply and export

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
Need walrus file apply, delete and export
**Solution:**
1. CLI support walrus file apply and delete
2. server side support walrus file apply and export 

**Related Issue:**
#1466

### CODE REVIEW

File: pkg/apis/environment/extension.go [hunk 1]
```diff
@@ -353,3 +358,131 @@ func (h Handler) RouteStop(req RouteStopRequest) error {
...
+	return name
+}
+
+func toResourceSpec(req RouteExportRequest, res *model.Resource) types.ResourceSpec {
+	spec := types.ResourceSpec{
+		Project: &types.MetadataName{
+			Name: req.Project.Name,
+		},
+		Environment: &types.MetadataName{
+			Name: req.EnvironmentQueryInput.Name,
```
Comment (gitlawr) [inline pkg/apis/environment/extension.go:471]: What happen if we export a resource from env A then apply to env B?

Comment (aiwantaozi) [reply pkg/apis/environment/extension.go:471]: While the exported file(A) includes the scope(project/environment), and the user applying with this file(Apply to B) will use the scope in the file. 
While the exported file doesn't include scope, apply will use the scope from API path params. To make the exported file apply/create to another env, prefer not to include the scope in the file.

**Review by gitlawr — APPROVED** [review]
LGTM

File: pkg/cli/formatter/yaml.go [hunk 2]
```diff
@@ -37,5 +38,10 @@ func (f *YamlFormatter) Format(resp *http.Response) ([]byte, error) {
 		return nil, err
 	}
 
+	// Response status is not 200.
```
Comment (alexcodelf) [inline pkg/cli/formatter/yaml.go:41]: How about compare status code before Unmarshal  body?

File: pkg/cli/formatter/json.go [hunk 3]
```diff
@@ -41,5 +42,10 @@ func (f *JsonFormatter) Format(resp *http.Response) ([]byte, error) {
 		return nil, err
 	}
 
+	// Response status is not 200.
```
Comment (alexcodelf) [inline pkg/cli/formatter/json.go:45]: How about compare status code before Unmarshal body?

File: pkg/cli/manifest/request.go [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+	if patchOpt == nil {
+		return success, failed, fmt.Errorf("not found patch operation for %s", group)
+	}
+
+	var (
+		results = make(chan result, 1)
+		all     = objs.All()
+	)
+
+	for _, obj := range all {
```
Comment (alexcodelf) [inline pkg/cli/manifest/request.go:64]: As our server has rate limit when object number is large will rate limit cause some patch failed?

File: pkg/apis/runtime/request_stream.go [hunk 5]
```diff
@@ -34,7 +34,10 @@ func (r RequestUnidiStream) Write(p []byte) (n int, err error) {
 
 // SendMsg sends the given data to client.
 func (r RequestUnidiStream) SendMsg(data []byte) error {
+	data = append(data, "\n\n"...)
```
Comment (alexcodelf) [inline pkg/apis/runtime/request_stream.go:37]: Why we need two line breaks here?

File: staging/utils/strs/rune.go [hunk 6]
```diff
@@ -52,3 +52,14 @@ func FillLeft(s string, w int) string {
 func FillRight(s string, w int) string {
 	return runewidth.FillRight(s, w)
 }
+
+// HasSuffix returns true if s ends with suffix.
+func HasSuffix(s string, suffixes ...string) bool {
+	for _, suffix := range suffixes {
+		if len(s) >= len(suffix) && s[len(s)-len(suffix):] == suffix {
```
[the remark this answers is not in the document]
Comment (aiwantaozi) [reply staging/utils/strs/rune.go:59]: The input suffixes mean could compare multiple suffixes and return true when matching one, just update the function comment.

File: pkg/apis/environment/extension.go [hunk 7]
```diff
@@ -353,3 +358,125 @@ func (h Handler) RouteStop(req RouteStopRequest) error {
...
 	})
 }
+
+func (h Handler) RouteApply(req RouteApplyRequest) error {
+	sj := session.MustGetSubject(req.Context)
+
+	token, err := auths.GetAccessToken(
+		req.Context, h.modelClient,
+		sj.ID, types.TokenKindAPI, types.WalrusOperationTokenName)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (aiwantaozi) [reply pkg/apis/environment/extension.go:368]: updated

File: pkg/cli/formatter/yaml.go (hunk 2 above)
Comment (aiwantaozi) [reply pkg/cli/formatter/yaml.go:41]: Unmarshal body before comparing the status is used for output the body while status is not 200.

File: pkg/cli/manifest/request.go (hunk 4 above)
Comment (aiwantaozi) [reply pkg/cli/manifest/request.go:64]: Now it will retry three times.

File: pkg/apis/runtime/request_stream.go (hunk 5 above)
Comment (aiwantaozi) [reply pkg/apis/runtime/request_stream.go:37]: This is used for separating each message, then could decode the event object from stream.

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**